hera_cal/abscal_funcs.py

```diff
@@ -661,29 +776,49 @@ def UVData2AbsCalDict(filenames, pol_select=None, pop_autos=True, return_meta=Fa
 
     return_meta : boolean, if True: also return antenna and unique frequency and LST arrays
 
+    filetype : string, filetype of data if datanames is a string, options=['miriad', 'uvfits']
+                can be ingored if datanames contains UVData objects.
+
+    pick_data_ants : boolean, if True and return_meta=True, return only antennas in data
+
     Output:
     -------
     if return_meta is True:
-        (data, flags, antpos, ants, freqs, times, pols)
+        (data, flags, antpos, ants, freqs, times, lsts, pols)
     else:
         (data, flags)
 
     data : dictionary containing baseline-pol complex visibility data
     flags : dictionary containing data flags
+    antpos : dictionary containing antennas numbers as keys and position vectors
+    ants : ndarray containing unique antennas
+    freqs : ndarray containing frequency channels (Hz)
+    times : ndarray containing LST bins of data (radians)
     """
-    # check filenames is a list
-    if type(filenames) is not list and type(filenames) is not np.ndarray:
-        if type(filenames) is str:
+    # check datanames is a list
+    if type(datanames) is not list and type(datanames) is not np.ndarray:
+        if type(datanames) is str:
             uvd = UVData()
-            uvd.read_miriad(filenames)
+            suffix = os.path.splitext(datanames)[1]
+            if filetype == 'uvfits' or suffix == '.uvfits':
+                uvd.read_uvfits(datanames)
+                uvd.unphase_to_drift()
+            elif filetype == 'miriad':
+                uvd.read_miriad(datanames)
+
         else:
-            uvd = filenames
+            uvd = datanames
```

Comment:
You might want to put some assert statements in here (like assert type is pyuvdata) so that it's easier for people to debug if they use this function incorrectly.

Reply:
added comments to make this clearer...